subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/annotation/resolver/TestKindExtractor.kt

```diff
@@ -0,0 +1,48 @@
+package com.avito.android.runner.annotation.resolver
+
+import com.avito.android.test.annotations.E2EStub
+import com.avito.android.test.annotations.E2ETest
+import com.avito.android.test.annotations.IntegrationTest
+import com.avito.android.test.annotations.ManualTest
+import com.avito.android.test.annotations.ScreenshotTest
+import com.avito.android.test.annotations.SyntheticMonitoringTest
+import com.avito.android.test.annotations.UIComponentStub
+import com.avito.android.test.annotations.UIComponentTest
+import com.avito.android.test.annotations.UnitTest
+import com.avito.report.model.Kind
+
+internal object TestKindExtractor {
+
+    fun extract(test: MethodStringRepresentation.Resolution): Kind {
+        val testTypes = arrayOf(
+            UIComponentTest::class.java,
+            E2ETest::class.java,
+            IntegrationTest::class.java,
+            ManualTest::class.java,
+            UIComponentStub::class.java,
+            E2EStub::class.java,
+            UnitTest::class.java,
+            ScreenshotTest::class.java,
+            SyntheticMonitoringTest::class.java
+        )
+        val testAnnotations = TestAnnotationExtractor.extract(test, *testTypes)
+
+        require(testAnnotations.isNotEmpty()) {
+            "Test $test has no type. Expected one of ${testTypes.joinToString { it.simpleName }}"
+        }
+        require(testAnnotations.size == 1) {
+            "Test $test has multiple types but must be only one: $testAnnotations"
+        }
+
+        return when (val testType = testAnnotations.first()) {
```

Comment:
What if the user added Kind annotation with a different type?

Reply:
This scenario is unsupported. A user can't add a custom Kind and test annotation for it in his app. 
This type of extensibility is out of the scope of this issue.